src/modm/architecture/interface/i2s_master.hpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2021, Marton Lednczki
+ * Copyright (c) 2022, Raphael Lehmann
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_INTERFACE_I2S_MASTER_HPP
+#define MODM_INTERFACE_I2S_MASTER_HPP
+
+#include <modm/processing/resumable.hpp>
+
+namespace modm
+{
+
+/**
+ * Interface for a I2S Master
+ *
+ * @author	Marton Ledneczki
+ * @ingroup	modm_architecture_i2s
+ */
+class I2sMaster : public ::modm::PeripheralDriver
```

Comment:
Should we somehow include in the name that this is an output device only? This interface does not describe a generic I²S master which could also receive audio or operate in time-multiplexed half-duplex mode. The terms master / slave do not imply a certain audio transfer direction, they only describe which device generates the clock.
If someone wanted to also implement input in the future there would be a naming issue.

Reply:
In order to not break compatibility when (later) implementing I2S full-duplex/input functionality I would like to keep the class name, but add _ONLY OUTPUT FUNCTIONALITY CURRENTLY IMPLEMENTED_ to the description.
(Slave (=clock receiver) would probably need another interface, but master with input you would probably just add more function/parameters to the current interface.)
What do you think?